lib/presentation/model/file/file_asset_entity.dart

```diff
@@ -0,0 +1,47 @@
+import 'dart:typed_data';
+
+import 'package:equatable/equatable.dart';
+import 'package:fluffychat/presentation/model/file/image_asset_entity.dart';
+import 'package:fluffychat/presentation/model/file/video_asset_entity.dart';
+import 'package:matrix/matrix.dart';
+import 'package:photo_manager/photo_manager.dart';
+
+class FileAssetEntity with EquatableMixin {
+
+  final AssetEntity assetEntity;
+
+  FileAssetEntity({required this.assetEntity});
+
+  Future<FileInfo?> toFileInfo() {
+    throw UnimplementedError('FileAssetEntity have not been implemented.');
+  }
+
+  Future<MatrixFile?> toMatrixFile() {
+    throw UnimplementedError('FileAssetEntity have not been implemented.');
+  }
+
+  String get messageType => throw UnimplementedError(
+    'FileAssetEntity have not been implemented.',
+  );
+
+  factory FileAssetEntity.createAssetEntity(AssetEntity asset) {
+    switch(asset.type) {
+      case AssetType.video:
+        return VideoAssetEntity(
+          assetEntity: asset,
+        );
+      case AssetType.image:
+        return ImageAssetEntity(
+          assetEntity: asset,
+        );
+      case AssetType.other:
+      case AssetType.audio:
+        throw UnimplementedError('unrecognized file type.');
```

Comment:
why not applied it as a binary file? `application/octet-stream`

Reply:
this is a handle for the audio file in the image picker, so I will do it in the next ticket. Video is enough :)

Reply:
Please create `DefaultFileAssetEntity` for this. 
`static const String File = 'm.file';`